Fix gcm validator

DianthuDia opened the pull request:
Pull request for gcm.

http://developer.android.com/google/gcm/gcm.html
- FIX: PayloadValidator

> 'data' field is a limit on the total size of the message (4kb).

However it validates total size of payload now.
- ADD: RegistrationIds validator

> It must contain at least 1 and at most 1000 registration IDs.

DianthuDia committed:
Fix payload validator

http://developer.android.com/intl/ja/google/gcm/gcm.html
'data' field is a limit on the total size of the message (4kb)

DianthuDia committed:
Add registration Ids validator

http://developer.android.com/intl/ja/google/gcm/gcm.html
It must contain at least 1 and at most 1000 registration IDs.

DianthuDia committed:
Fix 'registration_ids' column for MySQL

https://groups.google.com/forum/#!msg/android-gcm/q2PzJTP71TY/DNX4lvDu5X4J
The regID could grow up to 4k

ileitch commented on lib/generators/templates/add_gcm.rb:
```diff
@@ -34,7 +34,7 @@ def self.up
     add_column :rapns_notifications, :collapse_key, :string, :null => true
     add_column :rapns_notifications, :delay_while_idle, :boolean, :null => false, :default => false
-    add_column :rapns_notifications, :registration_ids, :text, :null => true
+    add_column :rapns_notifications, :registration_ids, :text, :null => true, :limit => 4*1024*1000
```
I don't want to add a db limit here. The size depends on the serialization format, which may change.

ileitch commented:
Other than the db limit, this looks good. Thanks.

DianthuDia committed:
Fix 'registration_ids' column type

Column type 'text' of MySQL represents < 65535 bytes.
Estimate of max size is 4k * 1000 = 4096000 bytes. (Serialized will be increasing.)

DianthuDia commented:
Thanks for reviewing.

I pushed some fix for 'mediumtext' when using mysql, mysql2.

ileitch commented:
Please explain the need to change the column type to medium text?

DianthuDia commented:
These commits change to store more data than before.

Column type 'text' of MySQL represents < 65535 bytes.
Estimate of max size is 4k \* 1000 = 4096000 bytes in speculation. (Serialized will be increasing.)
Estimate of real max size is 162 \* 1000 = 162000 bytes in my application.

ileitch commented:
Thanks :)
